rust/lance-index/src/vector/quantizer.rs

```diff
@@ -90,9 +92,9 @@ impl Quantizer {
 
     pub fn metadata_key(&self) -> &'static str {
         match self {
-            Self::Flat(fq) => fq.metadata_key(),
-            Self::Product(pq) => pq.metadata_key(),
-            Self::Scalar(sq) => sq.metadata_key(),
+            Self::Flat(_) => FlatQuantizer::metadata_key(),
+            Self::Product(_) => ProductQuantizerImpl::<Float32Type>::metadata_key(),
```

Comment:
do we need to support other float types?

Reply:
yeah, but won't be in this PR